Fix Cycle bug and update selected cycle in firebase

jpoist97 opened the pull request:
Fixed bug where cycles would not show up after reloading app. Also added functionality so that you can not select a cycle and it's updated in firebase.

jpoist97 committed:
Fix Cycle bug

ctperry0301 commented on components/Home/Home.jsx:
```diff
@@ -128,37 +128,36 @@ export default () => {
+  };
 
   const selectedCycle = (cycles.selectedCycleId !== undefined) && _.find(cycles.cycles, (cycle) => cycle.id === cycles.selectedCycleId);
   let cycleDetails;
```
You will get an index undefined error if you try to click on a left or right arrow on the home page if a cycle isn't set

ctperry0301 replied:
> You will get an index undefined error if you try to click on a left or right arrow on the home page if a cycle isn't set

Im also getting the same error if I try to create a cycle while not having a cycle currently selected (new account).

ctperry0301 reviewed (changes requested):
I dont think this code is working as expected for me, The cycles arent getting loaded in for me when I try and test it.

jpoist97 committed:
Merge branch 'main' of https://github.com/jpoist97/GainChanger into fix-cycle-bug

jpoist97 committed:
Fix cycles not loading on signup

Fixed bug where cycles wouldn't load on sign up. Cleaned up creating workouts and cycles. Prevent app from erroring when there is no selectedCycle.

jpoist97 commented:
Thanks for catching that Cole, I forgot to test making a new account. All the bugs should be fixed now

jpoist97 committed:
Make isCycleSelected prop required

jpoist97 committed:
Fix ExerciseDetails missing key prop

ctperry0301 reviewed (approved):
Good Stuff!
